### DISCUSSION TITLE
oo/abstract

### DISCUSSION THREAD

**Discussion by giscus** [itwanger/tobebetterjavaer-giscus#111] [Announcements]
# oo/abstract

Java程序员进阶之路，小白的零基础Java教程，认真聊聊 Java抽象类

https://tobebetterjavaer.com/oo/abstract.html

**Comment by SleepInthe** [comment]
没学io呢就整这种代码，纯纯让新手眼瞎

**Reply by DylanLJC** [reply]
同意，新人看不懂，但是，可以作为一个例子，希望作者可以标注出每行代码的含义。以及在哪儿个章节对应

**Reply by 2258324319** [reply]
啥玩意该学啥玩意不该学都整不明白还学啥。。。

**Reply by itwanger** [reply]
不至于吧，学啥都讲究一个前后顺序，就真没法学了，遇到不会的就先放一放，后面再来学，就像你做阅读理解，不可能所有单词都能会啊，至少得根据上下文自己去理解一下，不然你咋学？

**Reply by 0319cc** [reply]
可以二刷啊，二刷看这种代码就豁然开朗了

**Reply by icc0216** [reply]
吃个饭还得把饭一口一口喂你嘴里

**Comment by DylanLJC** [comment]
抽象类其实是可以被实例化的，但是它的实例化方式并不是通过普通的new方式来创建对象，而是通过父类的应用来指向子类的实例间接地实现父类的实例化，因为子类在实例化之前，一定会先实例化它的父类。这样创建了继承抽象类的子类对象，也就把其父类（抽象类）给实例化了。

**Reply by DylanLJC** [reply]
参考资料：https://blog.csdn.net/when_less_is_more/article/details/78004363
我觉得这样解释更合理一点

**Reply by itwanger** [reply]
厉害了！

**Reply by itwanger** [reply]
不过，硬说抽象类可以实例化的话，感觉比较牵强。也容易误导人。

**Reply by stardustman** [reply]
胡说八道。自己 heapdump 分析一下，看看抽象类到底能不能实例化。图片如下：
![image](https://github.com/itwanger/tobebetterjavaer-giscus/assets/22998204/abe3afcd-203c-437b-a044-ca21a3b2ad3c)
抽象类虽然不能实例化，但是并不意味着你抽象类的方法，子类就不可以访问了，子类的 class file 里可是有 super class 的，知道 super class 不就能调用它的方法了么。

**Comment by pittosporum-seu** [comment]
勘误：最后一句““呼。。。。。”一个大大的眼圈飘散开来，又是愉快的一天~” --> "“呼。。。。。”一个大大的烟圈飘散开来，又是愉快的一天~"

**Comment by alvin-succ** [comment]
我的项目咋没有resource目录

**Reply by 0319cc** [reply]
学一下maven
